iree/compiler/Dialect/Flow/Transforms/GenerateDispatchTags.cpp

```diff
@@ -0,0 +1,79 @@
+#include <utility>
+
+#include "iree/compiler/Dialect/Flow/IR/FlowDialect.h"
+#include "iree/compiler/Dialect/Flow/IR/FlowOps.h"
+#include "iree/compiler/Dialect/Flow/Transforms/PassDetail.h" 
+#include "iree/compiler/Dialect/Flow/Transforms/Passes.h"
+#include "iree/compiler/Dialect/Util/IR/UtilDialect.h"
+#include "iree/compiler/Dialect/Util/IR/UtilOps.h"
+#include "iree/compiler/Dialect/Shape/IR/Builders.h" 
+#include "iree/compiler/Dialect/Shape/IR/ShapeOps.h"
+#include "llvm/Support/Debug.h"
+#include "mlir/Dialect/StandardOps/IR/Ops.h"
+#include "mlir/IR/Attributes.h"
+#include "mlir/IR/BlockAndValueMapping.h"
+#include "mlir/IR/Builders.h"
+#include "mlir/IR/BuiltinTypes.h"
+#include "mlir/IR/Diagnostics.h"
+#include "mlir/IR/Module.h"
+#include "mlir/IR/SymbolTable.h"
+#include "mlir/Pass/Pass.h"
+
+#define DEBUG_TYPE "iree-dispatch"
+
+namespace mlir {
+namespace iree_compiler {
+namespace IREE {
+namespace Flow {
+namespace {
+
+//Generate IR at executable scope.
+static std::string executableIRString(Operation *op) {
```

Comment:
Rather than print the entire op, we should do some summarization here, aiming for 1-3 lines, maybe <200 characters?

Using [this program](https://github.com/google/iree/blob/main/iree/test/e2e/models/bert_encoder_unrolled_fake_weights.mlir) we get output like [this mlir](https://gist.github.com/ScottTodd/91d85795a03f3109b85a7fb440d4b8a8) after `DeduplicateExecutables`.

I have something like this in mind:
```
serving_default__ireesm_dispatch_4 
  elementwise
serving_default__ireesm_dispatch_5 
  linalg.fill, linalg.matmul (tensor<?x384xf32>, tensor<384x?xf32>), elementwise
serving_default__ireesm_dispatch_6 
  linalg.fill, linalg.matmul (tensor<?x512xf32>, tensor<512x?xf32>), elementwise
serving_default__ireesm_dispatch_18
  linalg.batch_matmul (tensor<?x?x384xf32>, tensor<?x384x?xf32>)
```

(any elementwise ops or data manipulation ops will be pretty cheap or difficult to summarize, while the named linalg ops like matmul will be important for performance tuning)

We can start with simple heuristics once this pass has a place to run and the plumbing through to Tracy and other tools is done, but we should be careful about putting too much data in here, as that will increase binary size.

Reply:
Even just starting with a scan for `linalg.matmul` (and possibly a few other named ops: https://mlir.llvm.org/docs/Dialects/Linalg/) would be useful, since those are where most of our compute time should be going.